### PR TITLE
Add spm1d recipe

### PR DESCRIPTION
**Pull request by melund**
From the docs:
```
spm1d is a package for one-dimensional Statistical Parametric Mapping.

It uses random field theory to make statistical inferences regarding registered (normalized) sets of 1D measurements.
```

### CODE REVIEW

File: recipes/PyAV/meta.yaml [hunk 1]
```diff
@@ -1,50 +0,0 @@
-{% set name = "av" %}
```
Comment (jakirkham) [inline recipes/PyAV/meta.yaml]: In other words, please restore this file.

File: recipes/click-help-colors/meta.yaml [hunk 2]
```diff
@@ -1,43 +0,0 @@
-{% set name = "click-help-colors" %}
```
Comment (jakirkham) [inline recipes/click-help-colors/meta.yaml]: Same as above.

File: recipes/spm1d/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - numpy
+    - matplotlib
+
+test:
+  imports:
+    - spm1d
+
+about:
+  home: http://www.spm1d.org/
+  license: GPL
```
Comment (jakirkham) [inline recipes/spm1d/meta.yaml]: What version of GPL?

Comment (jakirkham) [inline recipes/spm1d/meta.yaml]: Also can we please include a license file from the source directory using `license_file`.

File: recipes/spm1d/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - scipy
+    - numpy
+    - matplotlib
```
Comment (jakirkham) [inline recipes/spm1d/meta.yaml]: Please drop dependencies that are not needed at build time.

**Comment by melund** [comment]
@jakirkham I will rebase it onto master again.

File: recipes/spm1d/meta.yaml (hunk 3 above)
Comment (melund) [inline recipes/spm1d/meta.yaml]: It can go in next release of `spm1d`.

File: recipes/spm1d/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+test:
+  imports:
+    - spm1d
+
+about:
+  home: http://www.spm1d.org/
+  license: GPL-3.0
+  #license_file: LICENSE.txt
```
Comment (jakirkham) [inline recipes/spm1d/meta.yaml]: Please add a link to PR ( https://github.com/0todd0000/spm1d/pull/57 ) above this comment as well.

**Comment by jakirkham** [comment]
One last easy change. Otherwise LGTM.

**Comment by melund** [comment]
@jakirkham Seems to fail on some windows targets due to a problem with qt4 and matplotlib1.5... 

```
  File "C:\Miniconda3\envs\_test\lib\site-packages\matplotlib\backends\__init__.py", line 32, in pylab_setup
    globals(),locals(),[backend_name],0)
  File "C:\Miniconda3\envs\_test\lib\site-packages\matplotlib\backends\backend_qt4agg.py", line 18, in <module>
    from .backend_qt5agg import FigureCanvasQTAggBase as _FigureCanvasQTAggBase
  File "C:\Miniconda3\envs\_test\lib\site-packages\matplotlib\backends\backend_qt5agg.py", line 16, in <module>
    from .backend_qt5 import QtCore
  File "C:\Miniconda3\envs\_test\lib\site-packages\matplotlib\backends\backend_qt5.py", line 31, in <module>
    from .qt_compat import QtCore, QtGui, QtWidgets, _getSaveFileName, __version__
  File "C:\Miniconda3\envs\_test\lib\site-packages\matplotlib\backends\qt_compat.py", line 137, in <module>
    from PyQt4 import QtCore, QtGui
ImportError: cannot import name 'QtCore'
```

Maybe related to this. https://github.com/ContinuumIO/anaconda-issues/issues/1068

**Comment by melund** [comment]
@jakirkham So, I got the tests to pass by pinning the test requirements to pyqt=4 for windows. 

So I think this is ready to go in.

File: recipes/spm1d/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+test:
+  imports:
+    - spm1d
+  requires:
+    - pyqt ==4.11.4  # [win]
+    - qt ==4.8.7  # [win]
+
+
+
```
Comment (jakirkham) [inline recipes/spm1d/meta.yaml]: Please drop to one blank line between the sections.

File: recipes/spm1d/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+about:
+  home: http://www.spm1d.org/
+  license: GPL-3.0
+  # https://github.com/0todd0000/spm1d/pull/57
+  #license_file: LICENSE.txt
```
Comment (jakirkham) [inline recipes/spm1d/meta.yaml]: xref: https://github.com/0todd0000/spm1d/pull/57

File: recipes/spm1d/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    
+  run:
+    - python
+    - setuptools
```
Comment (jakirkham) [inline recipes/spm1d/meta.yaml]: Do we need `setuptools` at run time?

File: recipes/spm1d/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    - scipy
+    - numpy
+    - matplotlib
+
+test:
+  imports:
+    - spm1d
+  requires:
+    - pyqt ==4.11.4  # [win]
+    - qt ==4.8.7  # [win]
```
Comment (melund) [inline recipes/spm1d/meta.yaml]: Great. I will try.

File: recipes/spm1d/meta.yaml (hunk 7 above)
Comment (melund) [inline recipes/spm1d/meta.yaml]: done

**Comment by melund** [comment]
@jakirkham.  Matplotlib stuff does indeed seem to have been solved. Great...

**Comment by jakirkham** [comment]
Great. Then we merge. 😄

**Pull request merged**